go.mod

```diff
@@ -12,34 +12,34 @@ require (
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.29.0
 	github.com/stretchr/testify v1.8.4
 	github.com/zclconf/go-cty v1.14.1
-	golang.org/x/exp v0.0.0-20230905200255-921286631fa9
-	golang.org/x/mod v0.12.0
+	golang.org/x/exp v0.0.0-20231006140011-7918f672742d
```

Comment:
I think we should also update the dependencies (`go get -u && go mod tidy`) on regular basis. Doing as part of release.